alacritty_terminal/src/renderer/rects.rs

```diff
@@ -36,16 +36,42 @@ impl RenderRect {
     }
 }
 
-struct RenderLine {
-    start: Point,
-    end: Point,
-    color: Rgb,
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct RenderLine {
+    pub start: Point,
+    pub end: Point,
+    pub color: Rgb,
 }
 
 impl RenderLine {
-    fn into_rect(self, flag: Flags, metrics: &Metrics, size: &SizeInfo) -> RenderRect {
-        let start_x = self.start.col.0 as f32 * size.cell_width;
-        let end_x = (self.end.col.0 + 1) as f32 * size.cell_width;
+    pub fn as_rects(&self, flag: Flags, metrics: &Metrics, size: &SizeInfo) -> Vec<RenderRect> {
```

Comment:
This is not `as_rects`, `as` prefix always assumes cheap conversion (e.g. `O(1)`). However this conversion has a linear complexity, so, please, change its name to avoid confusion.

Reply:
What would you propose? I just used `as` since it doesn't take `self`, but instead `&self`.

I don't think `into` is ever used without taking `self`. Though I generally agree with you.

Reply:
Maybe just `rects` or `get_rects`?